src/main/java/com/michelin/kafkactl/command/Delete.java

```diff
@@ -109,33 +111,56 @@ private List<Resource> parseResources(String namespace) {
         }
 
         // Generate a single resource with minimum details from input
-        return List.of(Resource.builder()
+        var builder = Resource.builder()
             .metadata(Metadata.builder()
                 .name(config.nameConfig.name)
                 .namespace(namespace)
                 .build())
-            .kind(optionalApiResource.get().getKind())
-            .build());
+            .kind(optionalApiResource.get().getKind());
+
+        if (config.nameConfig.version.isPresent()) {
+            builder = builder.spec(Map.of(VERSION, config.nameConfig.version.get()));
+        }
+        return List.of(builder.build());
     }
 
-    static class EitherOf {
+    /**
+     * By-name of by-file deletion config.
+     */
+    public static class EitherOf {
+        /**
+         * Configuration for deletion by name.
+         */
         @ArgGroup(exclusive = false)
         public ByName nameConfig;
 
+        /**
+         * Configuration for deletion by file.
+         */
         @ArgGroup(exclusive = false)
         public ByFile fileConfig;
     }
 
-    static class ByName {
+    /**
+     * By-name deletion config.
+     */
+    public static class ByName {
         @Parameters(index = "0", description = "Resource type.", arity = "1")
         public String resourceType;
 
         @Parameters(index = "1", description = "Resource name.", arity = "1")
         public String name;
+
+        @Option(names = {"-V", "--version"},
+                description = "Version of the resource to delete (only with <name> parameter).", arity = "0..1")
```

Comment:
Should we precise `Version of the schema to delete..."?

Reply:
yes